rpcs3/util/atomic.cpp

```diff
@@ -70,6 +74,110 @@ static inline bool ptr_cmp(const void* data, std::size_t size, u64 old_value, u6
 	return false;
 }
 
+#ifdef USE_STD
+namespace
+{
+	// Standard CV/mutex pair
+	struct cond_handle
+	{
+		std::condition_variable cond;
+		std::mutex mtx;
+
+		cond_handle() noexcept
+		{
+			mtx.lock();
+		}
+	};
+}
+
+// Arbitrary max allowed thread number
+static constexpr u32 s_max_conds = 512 * 64;
+
+static std::aligned_storage_t<sizeof(cond_handle), alignof(cond_handle)> s_cond_list[s_max_conds]{};
+
+alignas(64) atomic_t<u64> s_cond_bits[s_max_conds / 64];
+
+alignas(64) atomic_t<u32> s_cond_sema{0};
+
+static u32 cond_alloc()
+{
+	// Determine whether there is a free slot or not
+	if (!s_cond_sema.try_inc(s_max_conds + 1))
+	{
+		return 0;
+	}
+
+	// Diversify search start points to reduce contention and increase immediate success chance
+#ifdef _WIN32
+	const u32 start = GetCurrentProcessorNumber();
+#elif __linux__
+	const u32 start = sched_getcpu();
+#else
+	const u32 start = __rdtsc();
+#endif
```

Comment:
This block is not valid, since we are in an `#ifdef` context already, which is set to USE_STD, which excludes windows and linux already.

Reply:
Because of what I said, we should use correct function for current platform.